### PR TITLE
More flexible ItemGroup addition API

### PR DESCRIPTION
**Pull request by lukebemish**
This relatively small API adds a single method to register ItemGroups not created through the QuiltItemGroup builder.

### CODE REVIEW

**Review by OroArmor — APPROVED** [review]
@LambdAurora @EnnuiL Should we make a standard for import order so that prs dont change them every time?

**Comment by EnnuiL** [comment]
> @LambdAurora @EnnuiL Should we make a standard for import order so that prs dont change them every time?

That's called Checkstyle, and people should always use it

**Comment by lukebemish** [comment]
> > @LambdAurora @EnnuiL Should we make a standard for import order so that prs dont change them every time?
> 
> That's called Checkstyle, and people should always use it

Might want to add something to https://github.com/QuiltMC/quilt-standard-libraries/blob/1.19/CONTRIBUTING.md about it then - it doesn't appear to be autodetected by my IDE or anything

**Comment by EnnuiL** [comment]
> > > @LambdAurora @EnnuiL Should we make a standard for import order so that prs dont change them every time?
> > 
> > 
> > That's called Checkstyle, and people should always use it
> 
> Might want to add something to https://github.com/QuiltMC/quilt-standard-libraries/blob/1.19/CONTRIBUTING.md about it then - it doesn't appear to be autodetected by my IDE or anything

That's a fair point; We really need to document that

**Comment by LambdAurora** [comment]
You need the checkstyle Intellij plugin (if you're using Intellij, otherwise, good luck)
then import it as a code style
if you only import the file in the plugin settings it won't modify the auto-formatter

**Comment by lukebemish** [comment]
> You need the checkstyle Intellij plugin (if you're using Intellij, otherwise, good luck) then import it as a code style if you only import the file in the plugin settings it won't modify the auto-formatter

Sounds good. I can set that up to make sure it is styled right here. Once again, though, I'd add this to the CONTRIBUTING.md - it's not at all obvious to someone that hasn't made a PR here before

**Comment by lukebemish** [comment]
Huh. My IDE likes to insert the imports as I try to move them around to get them in the right order for checkstyle; checkstyle is also entirely uninformative besides "they're not ordered right". Lovely. I'll get that fixed shortly.

**Comment by lukebemish** [comment]
Alright, fixed the format stuff.

**Review by EnnuiL — APPROVED** [review]

File: library/item/item_group/src/testmod/java/org/quiltmc/qsl/item/group/test/ItemGroupTest.java [hunk 1]
```diff
@@ -50,6 +50,13 @@ public class ItemGroupTest implements ModInitializer {
 
 	private static final QuiltItemGroup[] MANY_GROUPS = IntStream.range(0, 20).mapToObj(i -> QuiltItemGroup.builder(new Identifier(NAMESPACE, "many_group_" + i)).build()).toArray(QuiltItemGroup[]::new);
 
+	private static final ItemGroup NOT_QUILT_ITEM_GROUP = QuiltItemGroup.register(index -> new ItemGroup(index, NAMESPACE+".test_not_quilt_group") {
```
Comment (EnnuiL) [inline library/item/item_group/src/testmod/java/org/quiltmc/qsl/item/group/test/ItemGroupTest.java]:
```suggestion
	private static final ItemGroup NOT_QUILT_ITEM_GROUP = QuiltItemGroup.register(index -> new ItemGroup(index, NAMESPACE + ".test_not_quilt_group") {
```

**Review by EnnuiL — CHANGES_REQUESTED** [review]
A microscopic nitpick:

**Review by EnnuiL — APPROVED** [review]

**Pull request merged**